modules/apps/29-fee/keeper/keeper.go

```diff
@@ -1,16 +1,23 @@
 package keeper
 
-/*
 import (
 	"github.com/tendermint/tendermint/libs/log"
 
 	"github.com/cosmos/cosmos-sdk/codec"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	capabilitykeeper "github.com/cosmos/cosmos-sdk/x/capability/keeper"
 	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
 	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
-	"github.com/cosmos/ibc-go/modules/apps/transfer/types"
+	"github.com/cosmos/ibc-go/modules/apps/29-fee/types"
+	channeltypes "github.com/cosmos/ibc-go/modules/core/04-channel/types"
 	host "github.com/cosmos/ibc-go/modules/core/24-host"
+	ibcexported "github.com/cosmos/ibc-go/modules/core/exported"
+)
+
+// Middleware must implement types.ChannelKeeper and types.PortKeeper expected interfaces
+// so that it can wrap IBC channel and port logic for underlying application.
+var (
+	_ types.ChannelKeeper = Keeper{}
+	_ types.PortKeeper    = Keeper{}
```

Comment:
middleware must also wrap ChannelKeeper and PortKeeper since middleware is a "2-way" wrapper between IBC and app